src/ws/cockpitauth.c

```diff
@@ -616,7 +618,8 @@ reply_authorize_challenge (CockpitSession *session)
   if (cockpit_authorize_type (session->authorization, &authorization_type) &&
       (g_str_equal (authorize_type, "*") || g_str_equal (authorize_type, authorization_type)))
     {
-      send_authorize_reply (session->transport, cookie, session->authorization);
+      const gchar *remote_peer = cockpit_creds_get_rhost (cockpit_web_service_get_creds (session->service));
```

Comment:
Sure wish (after the last round of conversations) we had an idiomatic way to write "might be NULL" here...